tensornetwork/sparse/lanczos_test.py

```diff
@@ -0,0 +1,160 @@
+  H = be.reshape(Hmat, (4, 4, 4, 4))
+  Hmat = np.array(Hmat)
+
+  def mv(x, backend):
+    net = TensorNetwork(backend=backend)
+    n1 = net.add_node(H)
+    n2 = net.add_node(x)
+    n1[2] ^ n2[0]
+    n1[3] ^ n2[1]
+    out_order = [n1[0], n1[1]]
+    result = n1 @ n2
+    result.reorder_edges(out_order)
+    return result.tensor
+
+  def vv(a, b, backend):
+    net = TensorNetwork(backend=backend)
+    n1 = net.add_node(net.backend.conj(a))
+    n2 = net.add_node(b)
+    n1[0] ^ n2[0]
+    n1[1] ^ n2[1]
+    result = n1 @ n2
+    return result.tensor
+
+  A = lanczos.LinearOperator(mv, ((4, 4), (4, 4)), dtype=dtype, backend=backend)
+  vv = lanczos.ScalarProduct(vv, dtype=dtype, backend=backend)
+  eta1, U1 = lanczos.eigsh_lanczos(A, vv)
+  eta2, U2 = np.linalg.eigh(Hmat)
+  v2 = U2[:, 0]
+  v2 = v2 / sum(v2)
+  v1 = np.reshape(U1[0], (D))
+  v1 = v1 / sum(v1)
+  np.testing.assert_allclose(eta1[0], min(eta2))
+  np.testing.assert_allclose(v1, v2)
+
+
+@pytest.mark.parametrize("backend,dtype", [
+    *list(zip(['numpy'] * len(np_dtypes), np_dtypes)),
+    *list(zip(['tensorflow'] * len(tf_dtypes), tf_dtypes)),
+    *list(zip(['pytorch'] * len(torch_dtypes), torch_dtypes)),
+    *list(zip(['jax'] * len(jax_dtypes), jax_dtypes)),
+])
+def test_eigsh_lanczos_2(backend, dtype):
+  be = backend_factory.get_backend(backend, dtype)
+  D = 16
+  np.random.seed(10)
+  tmp = be.randn((D, D))
+  Hmat = tmp + be.transpose(be.conj(tmp), (1, 0))
+  H = be.reshape(Hmat, (4, 4, 4, 4))
+  Hmat = np.array(Hmat)
+
+  def mv(x):
+    net = TensorNetwork(backend=backend)
+    n1 = net.add_node(H)
+    n2 = net.add_node(x)
+    n1[2] ^ n2[0]
+    n1[3] ^ n2[1]
+    out_order = [n1[0], n1[1]]
+    result = n1 @ n2
+    result.reorder_edges(out_order)
+    return result.tensor
+
+  def vv(a, b):
+    net = TensorNetwork(backend=backend)
+    n1 = net.add_node(net.backend.conj(a))
+    n2 = net.add_node(b)
+    n1[0] ^ n2[0]
+    n1[1] ^ n2[1]
+    result = n1 @ n2
+    return result.tensor
+
+  A = lanczos.LinearOperator(mv, ((4, 4), (4, 4)), dtype=dtype, backend=backend)
+  vv = lanczos.ScalarProduct(vv, dtype=dtype, backend=backend)
+  eta1, U1 = lanczos.eigsh_lanczos(A, vv)
+  eta2, U2 = np.linalg.eigh(Hmat)
+  v2 = U2[:, 0]
+  v2 = v2 / sum(v2)
+  v1 = np.reshape(U1[0], (D))
+  v1 = v1 / sum(v1)
+  np.testing.assert_allclose(eta1[0], min(eta2))
+  np.testing.assert_allclose(v1, v2)
+
+
+@pytest.mark.parametrize("backend,dtype", [
+    *list(zip(['numpy'] * len(np_dtypes), np_dtypes)),
+    *list(zip(['tensorflow'] * len(tf_dtypes), tf_dtypes)),
+    *list(zip(['pytorch'] * len(torch_dtypes), torch_dtypes)),
+    *list(zip(['jax'] * len(jax_dtypes), jax_dtypes)),
+])
+def test_LinearOperator_1(backend, dtype):
+  # pylint: disable=unused-argument
+  # pylint: disable=unused-argument
+  def mv(x, backend, c):
+    pass
+
+  with pytest.raises(ValueError):
+    lanczos.LinearOperator(mv, ((4, 4), (4, 4)), dtype=dtype, backend=backend)
+
```

Comment:
Please add a test for this code in a `jax.jit`. I do not believe this is going to work due to the `while converge` in the eigen solver.

Reply:
Need to do this still!